### PR TITLE
Bake unnumbered figures with caption

### PR DESCRIPTION
**Pull request by wilkus27**
Issue: https://github.com/openstax/cnx-recipes/issues/4020

### CODE REVIEW

**Comment by wilkus27** [comment]
@kswanson33 I added you to review, because my changes will affect your book.

File: lib/kitchen/directions/bake_figure.rb [hunk 1]
```diff
@@ -6,19 +6,15 @@ module BakeFigure
...
-            HTML
-          )
+          return if figure.has_class?('has-caption')
+
+          puts 'warning! add BakeUnnumberedFigures to recipe before BakeFigure'
           return
         end
 
+        figure.wrap(%(<div class="os-figure#{' has-splash' if figure.has_class?('splash')}">))
+
```
[the remark this answers is not in the document]
Comment (wilkus27) [reply lib/kitchen/directions/bake_figure.rb:17]: Actually never mind :smile: I just baked Precalculus with released kitchen version and it's already in this form.

[the remark this answers is not in the document]
Comment (kswanson33) [reply lib/kitchen/directions/bake_figure.rb:17]: I like the use of the warning to enforce directions being called in a certain order. What I'm confused about is why this code needs to be taken out of `BakeFigure` at all? It looks like `BakeUnnumberedFigures` does about the same thing. Or is something new added?

File: lib/kitchen/directions/bake_unnumbered_figures.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+              </div>
+            HTML
+          )
+        end
+      end
+
+      def self.bake_figure_with_caption(book:)
+        book.figures('$.unnumbered').search('figcaption').each do |figure_caption|
+          figure_caption.parent.add_class('has-caption')
+        end
```
Comment (kswanson33) [inline lib/kitchen/directions/bake_unnumbered_figures.rb:28]: I think these two loops can be combined into one, i.e.

```ruby
book.figures('$.unnumbered').each do |figure|
  next unless figure.first('figcaption')
 
  figure.add_class('has-caption')
  figure.wrap(%(<div class="os-figure#{' has-splash' if figure.has_class?('splash')}">))
  # ... remaining code ...
end

**Review by kswanson33 — COMMENTED** [review]
Just leaving a few questions, but since I'll be out, I'll leave the rest of the review up to the team!

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+* Create `BakeUnnumberedFigures` to cover cases with caption (minor)
```
Comment (kswanson33) [inline CHANGELOG.md:8]: also -- if this is changing how `BakeFigure` works, this will be a breaking change!

**Comment by wilkus27** [comment]
@kswanson33 I created separate PR for that: https://github.com/openstax/kitchen/pull/395. I found a solution that is working. I assigned you there for review. Previously I didn't add this part:

image (20)

And that's why I got an error :face_palm: 
I'm going to close this PR.

**Pull request closed without merging**